### ISSUE TITLE
[FR] Auto-populate model_type on LoggedModel from flavor module when not explicitly provided

### ISSUE THREAD

**Issue by jayatijain** [mlflow/mlflow#24254]
> [!WARNING]
> Before submitting a PR, please make sure that:
> - A maintainer has triaged this issue and applied the `ready` label
> - This issue has no assignee
> - No duplicate PR exists
>
> PRs not meeting these requirements may be automatically closed.

### Willingness to contribute

Yes. I can contribute this feature independently.

### Proposal Summary

When mlflow.<flavor>.log_model() is called without an explicit model_type= argument, the resulting LoggedModel entity always has model_type: "", even though the framework (sklearn, pytorch, xgboost, etc.) is fully known at call time.

with mlflow.start_run():
    lr = LogisticRegression()
    lr.fit(X, y)
    mlflow.sklearn.log_model(lr, "model")

# Search for the logged model:
client.search_logged_models(experiment_ids=["1"])
# → LoggedModel.model_type == ""

model_type is the only entity-level field designed to identify the model framework. Everything else — the actual flavor dict, serialization format, Python version — lives inside the MLmodel artifact file and is only accessible by downloading it:

info = mlflow.models.get_model_info("models:/m-<uuid>")
info.flavors  # {"sklearn": {...}, "python_function": {...}}

This means any consumer that needs to know the model framework (for routing, filtering, UI display, downstream serving decisions) must download an artifact per model — which is impractical when searching across many logged models.

---
Describe the solution you'd like

Each flavor's log_model() should default model_type to the flavor name when the caller does not supply it. For example:

# mlflow/sklearn/__init__.py
def log_model(sk_model, artifact_path, ..., model_type=None, ...):
    return Model.log(
        ...,
        model_type=model_type or "sklearn",
        ...
    )

The same one-line change applies to all flavor modules: pytorch, tensorflow, xgboost, lightgbm, transformers, langchain, keras, onnx, catboost, h2o, dspy, statsmodels, spacy, llama_index, etc.

Backward compatibility: fully preserved. If the user already passes model_type="my-custom-type" explicitly, that value is used unchanged. The default only fires when model_type is None.

---
Describe alternatives you've considered

Alternative 1 — Use mlflow.models.get_model_info()
Works today, but downloads the MLmodel artifact on every call. For a search returning 50+ models this is 50+ artifact round-trips. Not viable for anything performance-sensitive.

Alternative 2 — Add a flavors field to LoggedModelInfo and a new SqlLoggedModelFlavor table
A heavier change. model_type already exists and is persisted by SqlAlchemyStore — we just need to fill it. Adding a new table and proto field is unnecessary for the core use case of "which framework produced this model."

---
Additional context

Investigated against mlflow 3.10.1:

- MlflowClient.create_logged_model() — no flavors parameter, model_type is user-supplied and defaults to None
- _tracking_service.client.create_logged_model() — has an internal flavor parameter, but it is explicitly commented: "only used for telemetry purposes, and does not affect the logged model" — so the framework name is already present at call time, just not written to the entity
- SqlAlchemyStore.create_logged_model() — already accepts and persists model_type correctly; no store changes needed
- SqlLoggedModelFlavor — does not exist; the LoggedModelInfo proto has no flavors field
- LoggedModelInfo proto fields: model_id, experiment_id, name, artifact_uri, status, model_type, source_run_id, tags, registrations — model_type is there, just always empty

The fix is entirely in the flavor layer — no proto changes, no migrations, no store changes.

### Motivation

> #### What is the use case for this feature?

> #### Why is this use case valuable to support for MLflow users in general?

> #### Why is this use case valuable to support for your project(s) or organization?

> #### Why is it currently difficult to achieve this use case?

### What machine learning domain(s) is this feature request about?

- [ ] `domain/genai`: LLMs, Agents, and other GenAI-related use cases
- [ ] `domain/classical-ml`: Traditional machine learning, such as linear regression.
- [ ] `domain/deep-learning`: Deep learning and neural networks.
- [ ] `domain/platform`: MLflow platform foundation, not specific to a particular machine learning domain.

### What area(s) of MLflow is this feature request about?

- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/scoring`: MLflow model serving, deployment tools, Spark UDFs
- [ ] `area/evaluation`: MLflow model evaluation features, evaluation metrics, and evaluation workflows
- [ ] `area/prompt`: MLflow prompt engineering features, prompt templates, and prompt management
- [ ] `area/tracing`: MLflow Tracing features, tracing APIs, and LLM tracing functionality
- [ ] `area/gateway`: MLflow AI Gateway client APIs, server, and third-party integrations
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/uiux`: Front-end, user experience, plotting
- [ ] `area/docs`: MLflow documentation pages

**Comment by arushiiii18** [comment]
Hi, I'd be interested in working on this issue once it's ready for contributions. Since I noticed it doesn't yet have the `ready` label, I wanted to check whether maintainers anticipate this approach being accepted and whether this would be a good issue to pick up after triage.

**Comment by faizanwasif** [comment]
Hi @jayatijain , this looks like a well-scoped fix. I'd like to pick it up and submit a PR. I've traced the relevant code paths (SqlAlchemyStore.create_logged_model() already persists model_type correctly, and the LoggedModelInfo proto has the field but it's just never populated by the flavor modules), so the change looks contained to adding model_type or "<flavor_name>" defaults in each flavor's log_model(), as you described.

Could a maintainer confirm this is ready to be picked up and apply the ready label?
